wicket/src/ui/panes/overview.rs

```diff
@@ -439,3 +444,429 @@ impl Control for InventoryView {
+                        },
+                    ]
+                    .into(),
+                );
+                if let Some(caboose) =
+                    sp.rot().and_then(|r| r.caboose_a.as_ref())
+                {
+                    append_caboose(&mut spans, caboose);
+                } else {
+                    spans.push(
+                        vec![
+                            nest_bullet(),
+                            Span::styled("No further information", warn_style),
+                        ]
+                        .into(),
+                    );
+                }
+                spans.push(
+                    vec![bullet(), Span::styled("Slot B:", label_style)].into(),
+                );
+                spans.push(
+                    vec![
+                        nest_bullet(),
+                        Span::styled("Image SHA3-256: ", label_style),
+                        match slot_b_sha3_256_digest.as_ref() {
+                            Some(digest) => {
+                                Span::styled(digest.clone(), ok_style)
+                            }
+                            None => Span::styled("Unknown", warn_style),
+                        },
+                    ]
+                    .into(),
+                );
+                if let Some(caboose) =
+                    sp.rot().and_then(|r| r.caboose_b.as_ref())
+                {
+                    append_caboose(&mut spans, caboose);
+                } else {
+                    spans.push(
+                        vec![
+                            nest_bullet(),
+                            Span::styled("No further information", warn_style),
+                        ]
+                        .into(),
+                    );
+                }
+            }
+            RotState::CommunicationFailed { message } => {
+                spans.push(label.into());
+                spans.push(
+                    vec![
+                        bullet(),
+                        Span::styled(
+                            format!("Communication Failure: {message}"),
+                            bad_style,
+                        ),
+                    ]
+                    .into(),
+                );
+            }
+        }
+    } else {
+        label.push(Span::styled("Not available", bad_style));
+        spans.push(label.into());
+    }
+
+    // blank line separator
+    spans.push(Spans::default());
+
+    // Describe all components.
+    // TODO-correctness: component information will change with the IPCC /
+    // topology work that is ongoing; show our current stand-in for now.
+    let mut label = vec![Span::styled("Components: ", label_style)];
+    let components = sp.components();
+    if components.is_empty() {
+        label.push(Span::styled("Not available", bad_style));
+        spans.push(label.into());
+    } else {
+        spans.push(label.into());
+        for component_info in components {
```

Comment:
It's worth considering using a destructuring style here.

```rust
let SpComponentInfo { presence, description, device } = component_info;
```

This will make it so that if new fields get added to the component info, then that will be detected as a compile-time issue. That is probably desirable since we'd like to show all component info here.

Reply:
Good call; I was already intentionally ignoring a few fields, but making that explicit is a solid improvement. Added in a3e47cdf